### PR TITLE
Add IAM communication

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: src/communication/messagehandler.hpp [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    /**
+     * Destroys PB service.
+     */
+    virtual ~PBServiceItf() = default;
+};
+
+/**
+ * Protobuf service implementation.
+ */
+
```
Comment (mykola-kobets-epam) [inline src/communication/messagehandler.hpp:111]: redundant empty line

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

File: src/aosmbedtlsconfig.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+#ifndef AOSMBEDTLSCONFIG_H_
+#define AOSMBEDTLSCONFIG_H_
+
+#define MBEDTLS_NO_PLATFORM_ENTROPY
+#define MBEDTLS_ENTROPY_HARDWARE_ALT
+
+#include "../../aos_core_lib_cpp/include/aos/common/crypto/mbedtls/mbedtls_config.h"
+
+#endif
```
Comment (MykolaSolyanko) [inline src/aosmbedtlsconfig.h:16]: add empty line

File: src/utils/checksum.cpp [hunk 3]
```diff
@@ -25,18 +24,20 @@ aos::RetWithError<aos::StaticArray<uint8_t, aos::cSHA256Size>> CalculateSha256(c
         return digest;
     }
 
-    auto ret = tc_sha256_init(&s);
-    if (TC_CRYPTO_SUCCESS != ret) {
+    mbedtls_sha256_init(&ctx);
+
+    auto ret = mbedtls_sha256_starts(&ctx, 0);
+    if (ret != 0) {
```
Comment (MykolaSolyanko) [inline src/utils/checksum.cpp:30]: i think we need to use `mbedtls_sha256_free`

File: tests/communication/prj.conf [hunk 4]
```diff
@@ -19,7 +19,7 @@ CONFIG_NO_OPTIMIZATIONS=y
 
 CONFIG_NANOPB=y
 
-# Enable tiny crypt lib
+# Enable mbedtls
 
-CONFIG_TINYCRYPT=y
-CONFIG_TINYCRYPT_SHA256=y
+CONFIG_MBEDTLS=y
+CONFIG_MBEDTLS_BUILTIN=y
```
Comment (MykolaSolyanko) [inline tests/communication/prj.conf:25]: empty line

**Review by MykolaSolyanko — APPROVED** [review]

File: src/app/app.cpp [hunk 5]
```diff
@@ -78,3 +86,54 @@ aos::Error App::Init()
...
+             .IsNone()) {
+        return err;
+    }
+
+    if (!(err = mCertHandler.RegisterModule(mSMCertModule)).IsNone()) {
+        return err;
+    }
+
+    return aos::ErrorEnum::eNone;
+}
```
Comment (MykolaSolyanko) [inline src/app/app.cpp:139]: add empty line

**Review by MykolaSolyanko — APPROVED** [review]

**Pull request merged**